### ISSUE TITLE
[Feature Request] Provide a UI to visualize container vulnerabilities

### ISSUE THREAD

**Issue by cdchris12** [goharbor/harbor#10496]
When trying to identify which containers within our registry contain vulnerabilities, it is frustrating to have to check every individual tag, within every individual registry, within every individual project. It's entirely infeasible to check every single container vulnerability report using the current UI framework for doing so.

Ideally, we would like to see a UI which would show the status of all container scans, scoped according to a user's permissions to view specific projects. This UI would allow users concerned with maintaining container security to get a high level view of the actual contents of these scans across all the projects they have access to. This data lends itself well to helping users determine which CVEs are worth allocating time to resolving due to their widespread occurrence across all their stored container images

**Comment by xaleeks** [comment]
I think maybe one view in each project is more reasonable.  We will think about this.  But this be easily retrievable via API as a workaround for now, let me know if you need help with that.

**Comment by gunboe** [comment]
This new feature sounds good for me too. A consolidated report and a critical CVE list of Images found will be nice to rectify or expunge the Image.

**Comment by dwctua** [comment]
A consolidated report with filter is crucial for operations.  A all-in-one on-line downloadable report to show all the vulnerabilities, white listing(scope, expiration) alone with artifact is very helpful. 
As side notes, it will be nice to have the CVE white list at image level to have better control.

**Comment by brondum** [comment]
+1 from here

**Comment by zyitingftnt** [comment]
> I think maybe one view in each project is more reasonable. We will think about this. But this be easily retrievable via API as a workaround for now, let me know if you need help with that.

@xaleeks I am interested to know which API to use in your solution? Thanks

**Comment by ikarlashov** [comment]
Any update on this? It feels like the whole scanning mechanism is useless without this feature. Indeed, it's not possible to check all images that are pushed by our developers. So we have 0 observability about vulnerabilities of our images.

**Comment by kirkanos** [comment]
Any updates on this? I really need this, a workaround via api doesn't really work for us.

**Comment by Mabchour** [comment]
Any updates on this.

**Comment by dale-c-anderson** [comment]
Figured out how to at least retrieve one repsitory's vulnerabilities from the API.

I assume one would need to loop over all the repos of all the projects, and also apply sorting to find only the latest vulnerability scan of a given repo.

To get a list of vulnerability scan result paths for a given repo,
```bash
curl -Ss --header "authorization: Basic $AUTH" -X GET "$HARBOR_SERVER/api/v2.0/projects/$PROJECT_NAME/repositories/$REPO_NAME/artifacts" | jq -r '.[].addition_links.vulnerabilities.href'
```
* *REPO_NAME needs to be escaped for slashes if it contains any - `foo-bar/ding` would be `foo-bar%2Fding`*.

To just arbitrarily accept the first result from the list, 
```bash
ARTIFACT_PATH=$(curl -Ss --header "authorization: Basic $AUTH" -X GET "${HARBOR_SERVER}/api/v2.0/projects/${PROJECT_NAME}/repositories/${REPO_NAME}/artifacts" | jq -r '.[0].addition_links.vulnerabilities.href')
```
To retrieve one scan of one repo of one project,
```bash
SCAN_RESULTS=$(curl -Ss --header "authorization: Basic $AUTH" -X GET "${HARBOR_SERVER}${ARTIFACT_PATH}")
```

**Comment by benediktberger** [comment]
We would love to integrate the scanning results in an external security overviews. Therefore I would need a link to display this information. As far as I could see from the rendered HTML there is no such possibility because the tooltip is rendered statically in the Harbor UI.
I was referring to a info badge like Dependency Track or GitLab provide, see https://docs.dependencytrack.org/integrations/badges/ or https://docs.gitlab.com/ee/user/project/badges.html for example.

**Comment by wy65701436** [comment]
We would like to make some improvements to the scan report view and @AllForNothing will gather requirements and investigate this part.

**Comment by matthiasbaldi** [comment]
@wy65701436, would there be a possibility to export the vulnerabilities as an SBOM to DependencyTrack?
So Harbor will scan (i.e. with Trivy) and then have an API to export that detected stuff as an valid SBOM to DependencyTrack to visualize everything?

**Comment by AllForNothing** [comment]
Fixed by #18942, closing it.
